### ISSUE TITLE
Installation complet but the service funkwhale-server won't start

### ISSUE THREAD

**Issue by JOduMonT** [YunoHost-Apps/funkwhale_ynh#149]
- Funkwhale version 1.1.2~ynh1
- Hardware: LXC
- OS: Debian 10.10
YunoHost: 4.2.6.1 stable
- Access: SSH, webadmin
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: No

### Steps to reproduce
I tried via the webadmin and via CLI `sudo yunohost app install funkwhale`

#### partial Log from the WebAdmin
```
The service funkwhale-server didn't fully executed the action start before the timeout.
Please find here an extract of the end of the log of the service funkwhale-server:
Jul 02 01:59:23 gunicorn[58932]: File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 58, in run
Jul 02 01:59:23 gunicorn[58932]: WSGIApplication("%(prog)s [OPTIONS] [APP_MODULE]").run()
Jul 02 01:59:23 gunicorn[58932]: File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/app/base.py", line 228, in run
Jul 02 01:59:23 gunicorn[58932]: super().run()
Jul 02 01:59:23 gunicorn[58932]: File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/app/base.py", line 72, in run
Jul 02 01:59:23 gunicorn[58932]: Arbiter(self).run()
Jul 02 01:59:23 gunicorn[58932]: File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 229, in run
Jul 02 01:59:23 gunicorn[58932]: self.halt(reason=inst.reason, exit_status=inst.exit_status)
Jul 02 01:59:23 gunicorn[58932]: File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 342, in halt
Jul 02 01:59:23 gunicorn[58932]: self.stop()
Jul 02 01:59:23 gunicorn[58932]: File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 393, in stop
Jul 02 01:59:23 gunicorn[58932]: time.sleep(0.1)
Jul 02 01:59:23 gunicorn[58932]: File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 242, in handle_chld
Jul 02 01:59:23 gunicorn[58932]: self.reap_workers()
Jul 02 01:59:23 gunicorn[58932]: File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 525, in reap_workers
Jul 02 01:59:23 gunicorn[58932]: raise HaltServer(reason, self.WORKER_BOOT_ERROR)
Jul 02 01:59:23 gunicorn[58932]: gunicorn.errors.HaltServer:
Jul 02 01:59:23 gunicorn[58932]: [2021-07-02 01:59:23 +0000] [58944] [INFO] Worker exiting (pid: 58944)
Jul 02 01:59:23 systemd[1]: funkwhale-server.service: Main process exited, code=exited, status=1/FAILURE
Jul 02 01:59:23 systemd[1]: funkwhale-server.service: Failed with result 'exit-code'.
```

#### Log from the CLI

```admin@ynh:~$ sudo yunohost app install funkwhale
Choose the domain where this app should be installed [cloud.jdmnt.co | om.jdmnt.co | zik.jdmnt.co] (default: om.jdmnt.co): zik.jdmnt.co
Choose an administrator user for this app [jond] (default: jond): 
Should this app be exposed to anonymous visitors? [yes | no] (default: yes): 
Info: Installing funkwhale...
Info: [+...................] > Validating installation parameters...
Info: [#+..................] > Storing installation settings...
Info: [##+.................] > Finding an available port...
Info: [###+................] > Installing dependencies...
Info: [####+...............] > Configuring system user...
Info: [#####+..............] > Creating a PostgreSQL database...
Info: [######+.............] > Setting up source files...
Info: [#######+............] > Configuring NGINX web server...
Info: [########++..........] > Adding a config file...
Info: [##########+.........] > Configuring a systemd service...
Info: [###########+........] > Installing Python dependencies...
Warning: Cache entry deserialization failed, entry ignored
Warning:   Running command git clone -q https://github.com/agateblue/requests-http-signature.git /tmp/pip-req-build-s5ey6ppa
Warning:   Running command git checkout -b signature-header-support --track origin/signature-header-support
Warning:   Switched to a new branch 'signature-header-support'
Warning:   Branch 'signature-header-support' set up to track remote branch 'signature-header-support' from 'origin'.
Info: [############+.......] > Building funkwhale...
Warning: 2021-07-02 03:24:52,971 funkwhale_api.config INFO     Loaded env file at /var/www/funkwhale/config/.env
Warning: 2021-07-02 03:24:52,972 funkwhale_api.config INFO     Running with the following plugins enabled: funkwhale_api.contrib.scrobbler, funkwhale_api.contrib.listenbrainz                                                                                                
Warning: 2021-07-02 03:25:02,925 funkwhale_api.config INFO     Loaded env file at /var/www/funkwhale/config/.env
Warning: 2021-07-02 03:25:02,926 funkwhale_api.config INFO     Running with the following plugins enabled: funkwhale_api.contrib.scrobbler, funkwhale_api.contrib.listenbrainz                                                                                                
Warning: 2021-07-02 03:25:04,353 funkwhale_api.config INFO     Loaded env file at /var/www/funkwhale/config/.env
Warning: 2021-07-02 03:25:04,354 funkwhale_api.config INFO     Running with the following plugins enabled: funkwhale_api.contrib.scrobbler, funkwhale_api.contrib.listenbrainz                                                                                                
Info: [#############+......] > Integrating service in YunoHost...
Info: [##############+.....] > Starting a systemd service...
Info: The service funkwhale-beat has correctly executed the action start.
Warning: (this may take some time)
Warning: The service funkwhale-server didn't fully executed the action start before the timeout.
Warning: Please find here an extract of the end of the log of the service funkwhale-server:
Warning: Jul 02 03:25:10 gunicorn[73106]:     sys.exit(run())
Warning: Jul 02 03:25:10 gunicorn[73106]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 58, in run                                                                                                                         
Warning: Jul 02 03:25:10 gunicorn[73106]:     WSGIApplication("%(prog)s [OPTIONS] [APP_MODULE]").run()
Warning: Jul 02 03:25:10 gunicorn[73106]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/app/base.py", line 228, in run                                                                                                                           
Warning: Jul 02 03:25:10 gunicorn[73106]:     super().run()
Warning: Jul 02 03:25:10 gunicorn[73106]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/app/base.py", line 72, in run                                                                                                                            
Warning: Jul 02 03:25:10 gunicorn[73106]:     Arbiter(self).run()
Warning: Jul 02 03:25:10 gunicorn[73106]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 229, in run                                                                                                                            
Warning: Jul 02 03:25:10 gunicorn[73106]:     self.halt(reason=inst.reason, exit_status=inst.exit_status)
Warning: Jul 02 03:25:10 gunicorn[73106]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 342, in halt                                                                                                                           
Warning: Jul 02 03:25:10 gunicorn[73106]:     self.stop()
Warning: Jul 02 03:25:10 gunicorn[73106]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 393, in stop                                                                                                                           
Warning: Jul 02 03:25:10 gunicorn[73106]:     time.sleep(0.1)
Warning: Jul 02 03:25:10 gunicorn[73106]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 242, in handle_chld                                                                                                                    
Warning: Jul 02 03:25:10 gunicorn[73106]:     self.reap_workers()
Warning: Jul 02 03:25:10 gunicorn[73106]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 525, in reap_workers                                                                                                                   
Warning: Jul 02 03:25:10 gunicorn[73106]:     raise HaltServer(reason, self.WORKER_BOOT_ERROR)
Warning: Jul 02 03:25:10 gunicorn[73106]: gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
Warning: Jul 02 03:25:10 systemd[1]: funkwhale-server.service: Main process exited, code=exited, status=1/FAILURE
Warning: Jul 02 03:25:10 systemd[1]: funkwhale-server.service: Failed with result 'exit-code'.
Info: The service funkwhale-worker has correctly executed the action start.
Info: [###############+....] > Configuring Fail2Ban...
Warning: <stdin>:7: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.                                                                                         
Warning: <stdin>:7: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.                                                                                         
Warning: <stdin>:7: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.                                                                                         
Warning: <stdin>:7: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.                                                                                         
Info: The service fail2ban has correctly executed the action reload-or-restart.
Info: [################+...] > Configuring permissions...
Info: [#################+..] > Reloading NGINX web server...
Info: [##################++] > Installation of funkwhale completed
Success! Installation completed
```

**Comment by asgardius** [comment]
I tried same Funkwhale version on a Raspberry Pi 3b+ using same Yunohost Version and i have the following log from journalctl
`-- Logs begin at Thu 2019-02-14 03:12:00 MST, end at Fri 2021-07-02 13:21:28 MST. --
jul 02 13:14:49 gunicorn[23094]:     django.setup()
jul 02 13:14:49 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/django/__init__.py", line 22, in setup
jul 02 13:14:49 gunicorn[23094]:     '/' if settings.FORCE_SCRIPT_NAME is None else settings.FORCE_SCRIPT_NAME
jul 02 13:14:49 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/django/urls/base.py", line 105, in set_script_prefix
jul 02 13:14:49 gunicorn[23094]:     _prefixes.value = prefix
jul 02 13:14:49 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/asgiref/local.py", line 113, in __setattr__
jul 02 13:14:49 gunicorn[23094]:     storage = self._get_storage()
jul 02 13:14:49 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/asgiref/local.py", line 83, in _get_storage
jul 02 13:14:49 gunicorn[23094]:     context_obj = self._get_context_id()
jul 02 13:14:49 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/asgiref/local.py", line 51, in _get_context_id
jul 02 13:14:49 gunicorn[23094]:     from .sync import AsyncToSync, SyncToAsync
jul 02 13:14:49 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/asgiref/sync.py", line 304, in <module>
jul 02 13:14:49 gunicorn[23094]:     class SyncToAsync:
jul 02 13:14:49 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/asgiref/sync.py", line 328, in SyncToAsync
jul 02 13:14:49 gunicorn[23094]:     loop = get_running_loop()
jul 02 13:14:49 gunicorn[23094]: RuntimeError: no running event loop
jul 02 13:14:49 gunicorn[23094]: [2021-07-02 20:14:49 +0000] [23106] [INFO] Worker exiting (pid: 23106)
jul 02 13:14:50 gunicorn[23094]: Traceback (most recent call last):
jul 02 13:14:50 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 209, in run
jul 02 13:14:50 gunicorn[23094]:     self.sleep()
jul 02 13:14:50 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 357, in sleep
jul 02 13:14:50 gunicorn[23094]:     ready = select.select([self.PIPE[0]], [], [], 1.0)
jul 02 13:14:50 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 242, in handle_chld
jul 02 13:14:50 gunicorn[23094]:     self.reap_workers()
jul 02 13:14:50 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 525, in reap_workers
jul 02 13:14:50 gunicorn[23094]:     raise HaltServer(reason, self.WORKER_BOOT_ERROR)
jul 02 13:14:50 gunicorn[23094]: gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
jul 02 13:14:50 gunicorn[23094]: During handling of the above exception, another exception occurred:
jul 02 13:14:50 gunicorn[23094]: Traceback (most recent call last):
jul 02 13:14:50 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/bin/gunicorn", line 8, in <module>
jul 02 13:14:50 gunicorn[23094]:     sys.exit(run())
jul 02 13:14:50 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 58, in run
jul 02 13:14:50 gunicorn[23094]:     WSGIApplication("%(prog)s [OPTIONS] [APP_MODULE]").run()
jul 02 13:14:50 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/app/base.py", line 228, in run
jul 02 13:14:50 gunicorn[23094]:     super().run()
jul 02 13:14:50 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/app/base.py", line 72, in run
jul 02 13:14:50 gunicorn[23094]:     Arbiter(self).run()
jul 02 13:14:50 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 229, in run
jul 02 13:14:50 gunicorn[23094]:     self.halt(reason=inst.reason, exit_status=inst.exit_status)
jul 02 13:14:50 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 342, in halt
jul 02 13:14:50 gunicorn[23094]:     self.stop()
jul 02 13:14:50 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 393, in stop
jul 02 13:14:50 gunicorn[23094]:     time.sleep(0.1)
jul 02 13:14:50 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 242, in handle_chld
jul 02 13:14:50 gunicorn[23094]:     self.reap_workers()
jul 02 13:14:50 gunicorn[23094]:   File "/var/www/funkwhale/virtualenv/lib/python3.7/site-packages/gunicorn/arbiter.py", line 525, in reap_workers
jul 02 13:14:50 gunicorn[23094]:     raise HaltServer(reason, self.WORKER_BOOT_ERROR)
jul 02 13:14:50 gunicorn[23094]: gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
jul 02 13:14:50 systemd[1]: funkwhale-server.service: Main process exited, code=exited, status=1/FAILURE
jul 02 13:14:50 systemd[1]: funkwhale-server.service: Failed with result 'exit-code'.`

**Comment by yalh76** [comment]
seems related to https://dev.funkwhale.audio/funkwhale/funkwhale/-/issues/1516

**Comment by JOduMonT** [comment]
I don't know to practice of YunoHost, but it might be a good practice to fix software version and only upgrade them after validation ;)
Cloudron, which of course is a paid software/service, do that, which allow them to offer very stable application.

**Comment by yalh76** [comment]
> 
> 
> I don't know to practice of YunoHost, but it might be a good practice to fix software version and only upgrade them after validation ;)
> Cloudron, which of course is a paid software/service, do that, which allow them to offer very stable application.

That's the case ... But YunoHost is volunteer time based....

The issue here come from python recently updated in Debian Buster.... Some deprecated functions have been removed, and Django, the framework used by Funkwhale, still use them ... would be solved quickly when Django and Funkwhale will update

**Comment by georgkrause** [comment]
> seems related to https://dev.funkwhale.audio/funkwhale/funkwhale/-/issues/1516

This reports a regression in develop, which is no issue in 1.1.2.

> The issue here come from python recently updated in Debian Buster.... Some deprecated functions have been removed, and Django, the framework used by Funkwhale, still use them ... would be solved quickly when Django and Funkwhale will update

What Debian, python and django version are you referring to?

**Comment by georgkrause** [comment]
> This reports a regression in develop, which is no issue in 1.1.2.

Turns out: I was completely wrong, it actually effects 1.1.2 if you do a fresh install without docker. Sorry for the wrong information.

Good news on the other site: It should be fixed with 1.1.4. Let me know if it works!
